ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListQueriesExecutor.java

```diff
@@ -40,25 +57,143 @@ private ListQueriesExecutor() { }
   ) {
     final ListQueries listQueries = statement.getStatement();
     if (listQueries.getShowExtended()) {
-      return Optional.of(new QueryDescriptionList(
-          statement.getStatementText(),
-          executionContext.getPersistentQueries().stream()
-              .map(QueryDescriptionFactory::forQueryMetadata)
-              .collect(Collectors.toList())));
+      return executeExtended(
+          listQueries,
+          statement,
+          sessionProperties,
+          executionContext,
+          serviceContext);
+    }
+
+    final List<RunningQuery> runningQueries = executionContext.getPersistentQueries()
+        .stream()
+        .map(q -> {
+          final KafkaStreamsStateCount kafkaStreamsStateCount = new KafkaStreamsStateCount();
+          kafkaStreamsStateCount.updateStateCount(q.getState(), 1);
+          return new RunningQuery(
+              q.getStatementString(),
+              ImmutableSet.of(q.getSinkName().text()),
+              ImmutableSet.of(q.getResultTopic().getKafkaTopicName()),
+              q.getQueryId(),
+              kafkaStreamsStateCount
+          );
+        }).collect(Collectors.toList());
+
+    final Map<String, RunningQuery> queryToRunningQuery = runningQueries.stream().collect(
+        Collectors.toMap(
+            query -> query.getId().toString(),
+            query -> query));
+
+    if (!sessionProperties.getInternalRequest()) {
+      final Set<HostInfo> hosts =
+          DiscoverRemoteHostsUtil.getRemoteHosts(
+              executionContext.getPersistentQueries(),
+              sessionProperties.getKsqlHostInfo());
+
+      hosts.forEach(hostInfo -> {
+        final KsqlEntityList response = serviceContext.getKsqlClient()
```

Comment:
what happens if there's an error with this? also I believe we have an `async` execute call on ksql client. Can we use that instead so that we don't handle them one at a time? (as your comment in the description mentions)